### PR TITLE
Use absolute target names in openssl

### PR DESCRIPTION
**Pull request by tapia**
Conan feature related: https://github.com/conan-io/conan/pull/10099

### CODE REVIEW

File: recipes/openssl/3.x.x/conanfile.py [hunk 1]
```diff
@@ -712,7 +712,7 @@ def _module_file_rel_path(self):
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "OpenSSL")
-        self.cpp_info.set_property("cmake_target_name", "OpenSSL")
+        self.cpp_info.set_property("cmake_target_name", "OpenSSL::OpenSSL")
```
Comment (SpaceIm) [inline recipes/openssl/3.x.x/conanfile.py:715]: doesn't exist in official FindOpenSSLcmake

Comment (czoido) [reply recipes/openssl/3.x.x/conanfile.py:715]:
```suggestion
```

Comment (Croydon) [reply recipes/openssl/3.x.x/conanfile.py:715]: Is `cmake_find_mode=both` defining both `OpenSSL` AND `OpenSSL:OpenSSL` targets? 
Wasn't the idea of those change to be more explicit?

Comment (czoido) [reply recipes/openssl/3.x.x/conanfile.py:715]: Sorry, I should have made that in another comment, It's two separate things, in one side removing that target that is not necessary and on the other forcing CMakeDeps to create the FindOpenSSL file

Comment (SpaceIm) [reply recipes/openssl/3.x.x/conanfile.py:715]: ah nice, interesting feature. so libcurl and sqlite3 also deserve `cmake_find_mode`.

File: recipes/openssl/3.x.x/conanfile.py [hunk 2]
```diff
@@ -712,7 +712,7 @@ def _module_file_rel_path(self):
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "OpenSSL")
-        self.cpp_info.set_property("cmake_target_name", "OpenSSL")
+        self.cpp_info.set_property("cmake_find_mode", "both")
```
Comment (SpaceIm) [inline recipes/openssl/3.x.x/conanfile.py:715]: And what kind of CMake variables names can we expect in the find file?

**Review by czoido — DISMISSED** [review]

File: recipes/openssl/3.x.x/conanfile.py (hunk 2 above)
Comment (czoido) [reply recipes/openssl/3.x.x/conanfile.py:715]: Yes, that won't work properly until we fix the variable names problem in the cmake modules that cmakedeps generates.

**Review by czoido — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**